resteasy-core/src/main/java/org/jboss/resteasy/plugins/interceptors/CacheControlFeature.java

```diff
@@ -74,4 +74,20 @@ protected CacheControl initCacheControl(NoCache value) {
             cacheControl.getNoCacheFields().add(field);
         return cacheControl;
     }
+
+    private Method findInterfaceBasedMethod(Class<?> root, Method method) {
+        if (method.getDeclaringClass().isInterface() || root.isInterface())
```

Comment:
The `method.getDeclaringClass().isInterface()` could throw an NPE. The `Class.getDeclaringClass()` may return `null`.

Reply:
I am not sure if it can return null. However `resourceInfo.getResourceMethod() or resourceInfo.getResourceClass()` could do that. I have updated the code accordingly to that

Reply:
Sorry, I think I mixed something up in my head :)